### ISSUE TITLE
[EDIT]: spam

### ISSUE THREAD

**Issue by IMalyugin** [getify/You-Dont-Know-JS#1486]
First of all, to start on a good note, I find this book overall very helpful and promising. It reveals a lot of hard to come by knowledge packed together in a short series, making it unique and educating.

But then there is a dark side. The books are crawling with author opinion, and each time he begins explaining his opinion, like with classes, implicity, coercion, it ends up very unprofessional.

**To author:**
Your hatered for classes, as justified as it might be, should not be spit it out on 100 of pages of your book in a deep blaming way, just to sum it up with “I don’t force my opinion on you, instead this book has all the information you need to create your own opinion”. That looks awfully subjective and immature.

~indexOf(...) looks more explicit than indexOf(...) !== -1 because it hides -1, you say? That is so wrong. To understand ~, you have to both know that result is -1, basic computer science and the fact that it applies to JS internal number representation.

“1 == [1] is totally alright”, you say? Because coercion is not at fault, one should blame valueOf implementation instead? - That is like saying “hey our planes are perfect, it’s the weather that makes them fall’. Blame the weather and just keep flying with us.

There is one key aspect your book clearly ignores, turning you into a factory for immature developers.
“You don’t write code for the compiler or yourself, you write code for other people”. So each time you put your silly praise for the implicity, it looks as if you are saying: “Please, by all means, name your functions and variables ‘a’, ‘b’, ‘c’,... because that is shorter! If somebody wants to figure out what your code does - they could look it up inside of your method.“

**Bottom line:**
As I wrote at the beginning, despite all that, I find the series very helpful, but it would be a lot better if a lot of aspects were written without forcing author's opinion so much on the reader. Especially if that opinion goes against the common opinion.
At the moment I find suggesting the series to others as a hard bargain. Because I have to suggest it in a twisted way such as this: "Please, read the series, they'll grow a deeper understanding of JS in you, but try to think of this book as a technical documentation and skip all the parts where author starts pointing out his opinion".
